### ISSUE TITLE
Conflation of manifest + payments seem unnecessary 

### ISSUE THREAD

**Issue by marcoscaceres** [WICG/paymentrequest#70]
> 1.) It allows us to leverage an existing specification that already defines a way of providing necessary branding elements (icons, colors) that are useful from a UI perspective. 

Although desirable, this doesn't equate to trust. Anyone can steal another person's manifest and icons. 

> 2.) It allows the User-Agent to fetch information about a payment app independent of an explicit install request from the payment app provider website.

I'm not sure this is so. The security model of manifest is bound to the Document that contains the manifest. It's not possible, or desirable, to use the manifest independently of the Document that contains that manifest. 

> Payment apps communicate with the service embedding the iFrame (e.g. the User-Agent) via postMessage.

Using an iframe seems problematic. You really want some kind of "overlay browsing context" that is distinct from an iframe: it would work like a "slide in" or "overlay" browsing context where payment can be securely made without needing to send messages back and forth. 

> 1.) We could implement a message passing mechanism wherein the first message the payment app posts back to the parent is whether or not UI is necessary. If not, the payment app could immediately post back the requisite information for payment. This is fine, but it still requires the use of iFrames and postMessage.

It seems you always want to show UI for payment confirmation. Using iframes and postMessage seems like a recipe for disaster. Please avoid this and just come up with a proper API.

**Comment by ianbjacobs** [comment]
Hi Marcos,

Adding link to the source of the text you are quoting:
https://github.com/w3c/webpayments/blob/b210a7b65e14818bed21334139828b0eeeb1417a/proposals/registration.md#defining-a-payment-app

Ian

**Comment by marcoscaceres** [comment]
Ah, thanks! Apologies for that.

**Comment by webpayments** [comment]
On 16 March 2016 at 03:35, ianbjacobs <notifications@github.com> wrote:

> Hi Marcos,
>
> Adding link to the source of the text you are quoting:
>
> https://github.com/w3c/webpayments/blob/b210a7b65e14818bed21334139828b0eeeb1417a/proposals/registration.md#defining-a-payment-app
>

A couple of questions regarding this document.

1. The term URL is used throughout.  Would URI be more appropriate here?

2. Would it not be a good idea to standardize JSON formats on the W3C REC,
JSON LD.  I appreciate manifests may be a slightly different case due to
existing widespread adoptions and standards.  However, JSON LD is also
starting to have widespread adoption and a unified serialization, that is
also a W3C REC, has benefits wrt interoperability, extensibility,
portability, testing, self description, and increasingly, network economics.

**Comment by marcoscaceres** [comment]
> 1. The term URL is used throughout.  Would URI be more appropriate here?

The term URI is deprecated. We don't use that anymore on the Web Platform.

> 2. Would it not be a good idea to standardize JSON formats on the W3C REC,
JSON LD.

That doesn't seem appropriate for the W3C. The W3C should focus on the platform, and not on formats that make use of standardized things (e.g., JSON). Note also that "W3C REC" doesn't always mean much. There are many RECs resulting from the W3C process being highjacked (intentionally or not) to push a particular agenda despite limited, or toy, implementations as proof of interoperability (e.g., GRDDL comes to mind). 

IMHO, it would be best of the W3C worked on API features that improve the platform. Protocols are best done at the IETF. Data formats based on underlying technology are best left to the community.

**Comment by melvincarvalho** [comment]
> The term URI is deprecated. We don't use that anymore on the Web Platform.

This statement left me slightly confused.  Regarding the TAG's [statement](URIs are the foundation of the value proposition for the Web)

> In The Architecture of the World Wide Web the TAG sets out the reasons why http: URIs are the foundation of the value proposition for the Web

Perhaps I am missing some context.  Is this something you disagree with?

**Comment by msporny** [comment]
We should really shut down this repo so folks stop raising issues here. The proper home for this issue is here:

https://github.com/w3c/browser-payment-api/issues

@marcoscaceres, would you mind migrating the issue over there and closing this one?

**Comment by dlongley** [comment]
I realize we may be migrating this, but I've got thoughts now, so I'm jotting them down...

@marcoscaceres,

> Using an iframe seems problematic. You really want some kind of "overlay browsing context" that is distinct from an iframe: it would work like a "slide in" or "overlay" browsing context where payment can be securely made without needing to send messages back and forth.

I agree with this. When an API call is made, a special browsing context should "slide in" and allow the user to make the appropriate selections, then the chosen Payment App handles the payment, and a acknowledgement/response is then handed back to the caller of the API.

> It seems you always want to show UI for payment confirmation. Using iframes and postMessage seems like a recipe for disaster. Please avoid this and just come up with a proper API.

I also agree with this. The iframe + postMessage approach works for building a polyfill (in fact, that's how we built the CG one), but we're trying to build the actual API here -- let's make this simpler so people don't have to implement, essentially, a polyfill. Isn't it going to be hard for developers to manage missed messages and we'll have to specify a communication protocol to avoid that? It sounds very clunky from a dev perspective, but perhaps I'm missing something. I'd rather just call the API and have it do the right thing.
